integration-tests/tests/day-janitor.js

```diff
@@ -17,30 +29,30 @@ const runJanitor = withUserToken(async ({ page, testUrl, token }) => {
   // ci test runs also sometimes leak workspaces.
   const workspaces = await page.evaluate(async () => await window.Ajax().Workspaces.list());
   const oldWorkspaces = _.filter(({ workspace: { name, createdDate } }) => {
-    const age = dateFns.differenceInDays(new Date(createdDate), new Date());
+    const age = getTimeDifference(new Date(createdDate), new Date());
     // console.info(`${namespace} ${name}, age ${age} days`)
-    return _.startsWith(testWorkspaceNamePrefix, name) && age > olderThanDays;
+    return _.startsWith(testWorkspaceNamePrefix, name) && age > olderThanTimeUnits;
   }, workspaces);
 
   console.log(
-    `Attempting to delete ${oldWorkspaces.length} workspaces with prefix "${testWorkspaceNamePrefix}" created more than ${olderThanDays} days ago.`
+    `Attempting to delete ${oldWorkspaces.length} workspaces with prefix "${testWorkspaceNamePrefix}" created more than ${olderThanTimeUnits} ${timeUnit} ago.`
   );
   console.log(`${_.filter({ workspace: { cloudPlatform: 'Azure' } }, oldWorkspaces).length} of the old workspaces are Azure workspaces.`);
   return Promise.all(
     _.map(async ({ workspace: { namespace, name, cloudPlatform } }) => {
       try {
         // Unlock the workspace in case it was left in a locked state during a test (locked workspaces can't be deleted).
-        await page.evaluate((namespace, name) => window.Ajax().Workspaces.workspace(namespace, name).unlock(), namespace, name);
-        await page.evaluate((namespace, name) => window.Ajax().Workspaces.workspace(namespace, name).delete(), namespace, name);
-        console.info(`Deleted old workspace: ${name}`);
+        await page.evaluate(async (namespace, name) => await window.Ajax().Workspaces.workspace(namespace, name).unlock(), namespace, name);
+        await deleteWorkspaceV2AsUser({ page, billingProject: namespace, workspaceName: name });
```

Comment:
With `deleteWorkspaceV2AsUser`, will we know if the workspace actually deletes? That is… does it poll to see if the delete succeeded, or does it just move on? I'm not entirely sure which I would prefer, so mostly just gathering data here.

Reply:
As it stands, this only polls the children, then sends the workspace delete req without waiting for success.